fc/code_generation.py

```diff
@@ -65,7 +65,9 @@ class DataInterpolation(Quantity):
     _next_id = 0  # Used to generate unique table IDs in code
 
     # Sympy annoyingly overwrites __new__
-    def __new__(cls, name, data, index_variable, *args, **kwargs):
+    def __new__(cls, name, data=None, index_variable=None, *args, **kwargs):
+        assert isinstance(name, str), str(name) +' '+ str(type(name))
```

Comment:
Why this line?

Reply:
I was trying to get it to behave like a proper quantity (it couldn't cope with the extra positional arguments), but ultimately I couldn't get it to work, so reverted & adding a work-around instead.